app/views/layouts/_filter.html.erb

```diff
@@ -0,0 +1,21 @@
+<div class="search">
+    <form>
+      <% if signed_in? && (current_page?(events_index_path) || current_page?(events_public_path))%>
+        <%= date_field :fecha, @date, class: 'field',required: true %>
+        <%= submit_tag "Filtrar Fecha", class: "filter" %>
+      <% end %>
+    </form>
+    <form>
+      <% if signed_in? && current_page?(events_index_path)%>
+        <%= select_tag(:privado, options_for_select([['Privado', true], ['Publico', false]]), class: 'check') %>
+        <%= submit_tag "Filtrar Tipo", class: "filter" %>
+      <% end %>
+
+    </form>
+      <% if signed_in? && current_page?(events_index_path)%>
+        <%= link_to 'Reiniciar', events_index_path, class: "filter" %>
+      <% end %>
```

Comment:
You are evaluating same the conditions two times, you could do it in a single evaluation. You can improve the logic of this code too.